js/Puck.js

```diff
@@ -1,15 +1,30 @@
 class Puck {
     #puckElement;
-    #rowLengths;
+    #rowLengths = [];
     #selectedCellIndexes;
     #onCellClickCb = [];
     #onCellHoverCb = [];
-    constructor(puckElement,rowLengths, selectedCellIndexes){
+    constructor(puckElement, rowLengths, selectedCellIndexes){
         this.#puckElement = puckElement;
-        this.#rowLengths =rowLengths;
         this.#selectedCellIndexes = selectedCellIndexes;
+        this.#rowLengths = rowLengths;
+    }
+    changeRowLengths(){
+        console.log(this.#rowLengths)
+        this.#rowLengths = [];
+        const button = document.getElementById('Re-draw');
```

Comment:
We certainly don't want to know about any buttons right in the puck. It's the outside code (that creates the puck) should know that it can draw and redraw pucks depending on a button and inputs.